src/Renci.SshNet/Common/SshData.cs

```diff
@@ -251,7 +251,14 @@ protected string[] ReadNamesList()
             {
                 var extensionName = ReadString(Ascii);
                 var extensionData = ReadString(Ascii);
-                result.Add(extensionName, extensionData);
+                if (result.ContainsKey(extensionName)) //if sftp extension is present multiple-times then keep the last one
```

Comment:
It's difficult to judge what the right behavior is for duplicate extension value.
Do you have a specification reference for SFTP extension to support a decision one way or another?
Can there be an approach similar to HTTP's multiple value for same header, which combines the values to a string with a known separator?
Or update the dictionary to be <string, IEnumerable<string>>